crates/acct-types/src/id.rs

```diff
@@ -0,0 +1,41 @@
+use std::{fmt, mem};
+
+/// Universal account identifier.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct AcctId([u8; 32]);
+
+/// Incrementally assigned account serial number.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct AcctSerial(u32);
```

Comment:
3 things:

1. I don't see anywhere in this PR the `AccountSerial` being used or incremented.
2. Why not calling `AccountSequenceNumber`?
3. You should implement a `increment` function that `+=1` since this is an opaque type with a non-pub inner value.